libraries/provider_slave_grants.rb

```diff
@@ -0,0 +1,45 @@
+class Chef
+  class Provider
+    # Used to setup permissions for MySQL replication user on master server
+    class MysqlmSlaveGrants < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      def action_create
+        grant_slave_privs
```

Comment:
I'm wondering if we should define `load_current_resource`, to see if we need to run `grant_slave_privs`